agent/include/content_analysis/sdk/analysis_agent.h

```diff
@@ -47,62 +60,99 @@ namespace sdk {
-  // Google Chrome.
+  // Retrieves a read-only reference to the content analysis request received
+  // from Google Chrome.
   virtual const ContentAnalysisRequest& GetRequest() const = 0;
 
-  // Retrieves a reference to the content analysis response that will be
-  // sent to Google Chrome as the verdict for the request of this session.
-  // The agent may modify this response in place before calling Send.
+  // Retrieves a writable reference to the content analysis response that will
+  // be sent to Google Chrome as the verdict for the request of this event.
+  // The agent may modify this response in place before calling Send().
   virtual ContentAnalysisResponse& GetResponse() = 0;
 
-  // Retrieves a reference to the Acknowledgement sent by Google Chrome after
-  // the agent responds with a verdict.
-  virtual const Acknowledgement& GetAcknowledgement() const = 0;
-
   // Send the verdict to Google Chrome.  Once this method is called further
   // changes to the response are ignored.
   virtual int Send() = 0;
 
  protected:
-  Session() = default;
-  Session(const Session& rhs) = delete;
-  Session(Session&& rhs) = delete;
-  Session& operator=(const Session& rhs) = delete;
-  Session& operator=(Session&& rhs) = delete;
+  ContentAnalysisEvent() = default;
+  ContentAnalysisEvent(const ContentAnalysisEvent& rhs) = delete;
+  ContentAnalysisEvent(ContentAnalysisEvent&& rhs) = delete;
+  ContentAnalysisEvent& operator=(const ContentAnalysisEvent& rhs) = delete;
+  ContentAnalysisEvent& operator=(ContentAnalysisEvent&& rhs) = delete;
+};
+
+// Agents should implement this interface in order to handle events as needed.
+//
+// OnBrowserConnected() and OnBrowserDisonnected() notify the agent when
+// instances of Google Chome start and stop.  The agent may perform any one-time
+// actions as required for these events.  The default action is to do nothing
+// for both events.  If the agent does not need perform any special actions
+// these methods do not need to be overridden.
+//
+// OnAnalysisRequested() notifies the agent of a new content analysis request
+// from Google Chrome.  The agent should perform the analysis and respond to
+// the event.  It is not required for the agent complete the analysis and
+// respond to before this callback returns.  The agent may pass the
+// ContentAnalysisEvent to a background task and respond when ready.  This
+// callback has no default action and agents must override it.
+//
+// OnResponseAcknowledged() notifies the agent that Google Chrome has received
+// the content analysis response and how it has handled it.
+class AgentEventHandler {
+ public:
+  AgentEventHandler() = default;
+  virtual ~AgentEventHandler() = default;
+
+  // Called when a new Google Chrome browser instance connects to the agent.
+  // This is always called before the first OnAnalysisRequested() from that
+  // browser.
+  virtual void OnBrowserConnected(const BrowserInfo& info) {}
```

Comment:
Does this and the following methods (OnBrowserDisconnected) need to return statuses to indicate the connection with Chrome is successfully connected/disconnected?

Reply:
These are callbacks after the connection or disconnection happens, to inform the agent that the event happened.  I'm not the SDK needs to know what the agent did with the event.  Do you have anything in mind?

Reply:
I see, I originally thought one needs to call the two methods to connect to/disconnect from browser. The current APIs LGTM.